pkg/core/resources.go

```diff
@@ -19,6 +19,8 @@ type (
 		KlothoConstructRef() []AnnotationKey
 		// Id returns the id of the cloud resource
 		Id() ResourceId
+		// Create modifies itself to be a completed instance of the resource given a set of metadata
+		Create(dag *ResourceGraph, metadata map[string]any) (Resource, error)
```

Comment:
I like that `Create` mutates (similar to `UnmarshalJSON`) but why does it return a `Resource`?

Also, would we expect every resource to implement this? If not, perhaps this could be a separate interface for resources to implement. Could save on a bunch of "Not Implemented" boilerplate.

This is a bit of a nit pick, but "metadata" doesn't really describe it well IMO - something like "parameters" or "arguments" feel closer to how it's used.